src/components/Product/Paging.jsx

```diff
@@ -1,21 +1,39 @@
 import styles from './styles/Paging.module.css';
+import { useNavigate, useLocation } from 'react-router-dom';
 export default function Paging({ totalPage, currentPage, setCurrentPage }) {
   const pageArr = new Array(totalPage).fill(0).map((_, i) => i + 1);
+  const navigate = useNavigate();
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+
+  const handlePageChange = (page) => {
+    setCurrentPage(page);
+    searchParams.set('page', page);
+    navigate(`${location.pathname}?${searchParams.toString()}`);
+  };
 
   let currentIdx = Math.floor((currentPage - 1) / 5);
   let currentPages = pageArr.slice(currentIdx * 5, currentIdx * 5 + 5);
 
   return (
     <div className={`${styles.Paging} ${totalPage <= 5 ? styles.hideArrow : ''}`}>
-      <button className={styles.arrow} onClick={() => currentPage > 1 && setCurrentPage(currentPage - 1)}>
+      <button 
+        className={`${styles.arrow} ${currentPage === 1 ? styles.disabled : ''}`} 
+        onClick={() => currentPage > 1 && handlePageChange(currentPage - 1)}
+        disabled={currentPage === 1}
+      >
         <PagingIconArrow reverse />
       </button>
       {currentPages.map((page) => (
-        <button key={page} className={currentPage === page ? styles.active : ''} onClick={() => setCurrentPage(page)}>
+        <button key={page} className={currentPage === page ? styles.active : ''} onClick={() => handlePageChange(page)}>
           {page}
         </button>
       ))}
-      <button className={styles.arrow} onClick={() => currentPage < totalPage && setCurrentPage(currentPage + 1)}>
+      <button 
+        className={`${styles.arrow} ${currentPage === totalPage ? styles.disabled : ''}`} 
+        onClick={() => currentPage < totalPage && handlePageChange(currentPage + 1)}
```

Comment:
💊 제안
특정 조건에서 해당 버튼을 disabled 처리하셨으니 `currentPage < totalPage` 조건 검사는 불필요합니다~

```suggestion
        onClick={() => handlePageChange(currentPage + 1)}
```